ui/src/authors/containers/__tests__/AssignOneDifferentProfileContainer.test.jsx

```diff
@@ -10,95 +11,149 @@ import {
   assignDifferentProfile,
 } from '../../../actions/authors';
 import AssignOneDifferentProfileContainer from '../AssignOneDifferentProfileContainer';
-import AssignOneDifferentProfileAction from '../../components/AssignOneDifferentProfileAction';
 
 jest.mock('react-router-dom', () => ({
   useParams: jest.fn().mockImplementation(() => ({
     id: 123,
   })),
 }));
 
+jest.mock('../../components/AssignOneDifferentProfileAction', () => {
+  const actual = jest.requireActual(
+    '../../components/AssignOneDifferentProfileAction'
+  );
+  return {
+    __esModule: true,
+    default: jest.fn((props) => <actual.default {...props} />),
+  };
+});
+
 jest.mock('../../../actions/authors');
 mockActionCreator(assignDifferentProfile);
 mockActionCreator(setPublicationSelection);
 mockActionCreator(clearPublicationSelection);
 
 describe('AssignOneDifferentProfileActionContainer', () => {
-  it('selects one paper and dispatches assignDifferentProfile', () => {
-    const store = getStore();
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('selects one paper and dispatches assignDifferentProfile with userCanNotClaimProfile=true', async () => {
+    const store = getStore({
+      user: fromJS({
+        data: {
+          recid: 456,
+        },
+      }),
+    });
     const paperRecordId = 12345;
-    const from = 123;
-    const to = 321;
     const userCanNotClaimProfile = true;
-    const wrapper = mount(
+    const { container } = render(
       <Provider store={store}>
         <AssignOneDifferentProfileContainer
           recordId={paperRecordId}
           userCanNotClaimProfile={userCanNotClaimProfile}
         />
       </Provider>
     );
-    const onAssign = wrapper
-      .find(AssignOneDifferentProfileAction)
-      .prop('onAssign');
-    onAssign({ from, to });
+
+    const dropdownTrigger = container.querySelector(
+      '[data-test-id="btn-claim"]'
```

Comment:
can we also utilize the `getByTestId` here ? also in all  the other places we use `document.querySelector` in the PR ?